collector/lib/CollectorConfig.cpp

```diff
@@ -360,4 +369,60 @@ std::ostream& operator<<(std::ostream& os, const CollectorConfig& c) {
          << ", enable_external_ips:" << c.EnableExternalIPs();
 }
 
+// Returns number of CPUs per buffer for the purpose of ring buffer allocation.
+// The value is adjusted based on the allowed total limit, e.g.:
+//
+// * total limit for ring buffers is 1Gi
+// * one buffer takes 8Mi
+// * maximum allowed number of buffers is 128
+// * total number of CPU cores is 150
+//
+// In this scenario, n_buffers = 150 / 1 > 128, thus the adjusted value will be
+// returned: ceil(150 / 128) = 2. Allocating one buffer per 2 CPU cores will
+// allow us to fit into the total limit: (150 / 2) * 8Mi < 1Gi.
+//
+unsigned int CollectorConfig::GetSinspCpuPerBuffer() const {
+  unsigned int n_buffers, max_n_buffers;
+
+  if (sinsp_buffer_size_ == 0) {
+    CLOG(WARNING) << "Trying to calculate cpu-per-buffer without"
+                     "initialized buffer size. Return raw value.";
+    return sinsp_cpu_per_buffer_;
+  }
+
+  if (host_config_.GetNumPossibleCPUs() == 0) {
+    CLOG(WARNING) << "Trying to calculate cpu-per-buffer without"
+                     "number of possible CPUs. Return raw value.";
+    return sinsp_cpu_per_buffer_;
+  }
+
+  // Round to the larger value, since one buffer will be allocated even if the
+  // last group of CPUs is less than sinsp_cpu_per_buffer_
+  n_buffers = std::ceil((host_config_.GetNumPossibleCPUs() / sinsp_cpu_per_buffer_));
```

Comment:
Both sides of the division in `std::ceil` are `unsigned int`, so that would take precedence and `std::ceil` wouldn't do much, right? You might want to cast one of them to force the result to be a float or double, or get rid of `std::ceil` altogether.